### PR TITLE
Add Feed 26: Create Affiliates

### PR DESCRIPTION
**Pull request by diego-binary**
Adds a new method for MyAffiliates account creation

```perl
use WebService::MyAffiliates;

my $aff = WebService::MyAffiliates->new(
    user => 'user',
    pass => 'awesome_pass',
    host => 'https://awesome.host.com/');

my $res = $aff->create_affiliate({
      'first_name'     => 'Chales',
      'last_name'     => 'Babbage',
      'date_of_birth' => '1871-10-18',
      'individual'    => 'individual',
      'phone_number'  => '+4412341234',
      'address'       => 'Some street',
      'city'          => 'Some City',
      'state'         => 'Some State',
      'postcode'      => '1234',
      'website'       => 'https://www.example.com/',
      'agreement'     => 1,
      'username'      => 'charles_babbage.com',
      'email'         => 'charles@babbage.com',
      'country'       => 'GB',
      'password'      => 's3cr3t',
      'plans'         => '2,4'
  });

die 'something failed' unless $res;

$res->{USERID};
```

### CODE REVIEW

File: t/create_affiliate.t [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    user => $ENV{MYAFFILIATES_USER},
+    pass => $ENV{MYAFFILIATES_PASS},
+    host => $ENV{MYAFFILIATES_HOST});
+
+my $res = $aff->create_affiliate({'incomplete' => 'params'});
+
+ok(!$res, 'Returns undef if incomplete parameters are passed');
+
+done_testing();
+
```
[the remark this answers is not in the document]
Comment (diego-binary) [reply t/create_affiliate.t:132]: Added mock library and tried to cover all the branches for create_affiliate

File: lib/WebService/MyAffiliates.pm [hunk 2]
```diff
@@ -45,25 +45,51 @@ sub __ua {
...
 }
 
+sub create_affiliate {
+    my $self = shift;
+    my $args = shift;
+    croak 'A hashref was expected as parameter' unless ref $args eq ref {};
+    my $parameters = {};
+    for my $arg (keys(${args}->%*)) {
+        $parameters->{'PARAM_' . $arg} = $args->{$arg};
+    }
```
Comment (felipe-binary) [inline lib/WebService/MyAffiliates.pm:63]:
```suggestion
    my $parameters = map +{ ('PARAM_' . $_ => $args->{$_} ) } keys $args->%*;
```
maybe more perlish

File: lib/WebService/MyAffiliates.pm [hunk 3]
```diff
@@ -45,25 +45,48 @@ sub __ua {
...
     my %args = @_ % 2 ? %{$_[0]} : @_;
     my $url  = Mojo::URL->new('/feeds.php?FEED_ID=1');
     $url->query(\%args) if %args;
     return $self->request($url->to_string);
 }
 
+sub create_affiliate {
+    my $self = shift;
+    my $args = shift;
+    croak 'A hashref was expected as parameter' unless ref $args eq ref {};
```
Comment (billmarriott-binary) [inline lib/WebService/MyAffiliates.pm:59]: This check isn't needed, the next line will die

Comment (diego-binary) [reply lib/WebService/MyAffiliates.pm:59]: I think is better to be explicit about the error.

**Pull request merged**